### PR TITLE
DT-1376: ✨ Use new auth search to find external users

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: backend/api/oauthApi.js [hunk 1]
```diff
@@ -3,13 +3,26 @@ const axios = require('axios')
...
+const contextProperties = require('../contextProperties')
 
 const AuthClientErrorName = 'AuthClientError'
 const AuthClientError = (message) => ({ name: AuthClientErrorName, message, stack: new Error().stack })
 
-const encodeQueryString = (input) => encodeURIComponent(input)
 const apiClientCredentials = (clientId, clientSecret) => Buffer.from(`${clientId}:${clientSecret}`).toString('base64')
 
+const processPageResponse = (context) => (response) => {
+  if (response.body.pageable) {
```
Comment (petergphillips) [inline backend/api/oauthApi.js:14]: first time we've done paging here - copied from prisonstaffhub

File: backend/controllers/authUserMaintenance.js [hunk 2]
```diff
@@ -43,13 +43,11 @@ const authUserMaintenanceFactory = (oauthApi) => {
 
     await handleClientError(
       async () => {
-        const response = nameFilter.includes('@')
-          ? await oauthApi.userSearch(res.locals, { nameFilter })
-          : [await oauthApi.getUser(res.locals, { username: nameFilter })]
+        const response = await oauthApi.userSearch(res.locals, { nameFilter })
```
Comment (petergphillips) [inline backend/controllers/authUserMaintenance.js:46]: now just call user search

File: integration-tests/integration/login/authuser.spec.js [hunk 3]
```diff
@@ -49,9 +49,9 @@ context('External user functionality', () => {
   it('Should add and remove a role from a user', () => {
     const results = searchForUser()
 
+    cy.task('stubAuthGetUsername')
     cy.task('stubAuthUserRoles')
     cy.task('stubAuthUserGroups')
-    cy.server()
```
Comment (petergphillips) [inline integration-tests/integration/login/authuser.spec.js:54]: not needed for cypress 6

File: integration-tests/plugins/index.js [hunk 4]
```diff
@@ -22,6 +22,7 @@ module.exports = (on) => {
     stubVerifyToken: (active = true) => tokenverification.stubVerifyToken(active),
     stubLoginPage: auth.redirect,
     stubAuthUsernameSearch: auth.stubAuthUsernameSearch,
+    stubAuthGetUsername: auth.stubAuthGetUsername,
```
Comment (petergphillips) [inline integration-tests/plugins/index.js:25]: we were previously using username search for the get username call, but now have had to split out

**Review by Mjwillis — APPROVED** [review]
LGTM

**Pull request merged**